### DISCUSSION TITLE
Recommended update policy

### DISCUSSION THREAD

**Discussion by alexpdp7** [siderolabs/talos#7242] [Q&A] [answered]
Is there a recommended update policy?

Should one track Talos versions, then update minor K8S versions within that?

For example, I'm running now Talos 1.4.4, which ships with K8S 1.27.1.

Should I update to any 1.27.x? Should I update to 1.28 when it appears?

**Comment by alexpdp7** [comment]
(Asking this because my [update monitor](https://github.com/alexpdp7/talos-check) now reports:

```
{
 "cluster_kubernetes_version": "v1.27.1",
 "cluster_talos_versions": [
  "v1.4.4"
 ],
 "available_kubernetes_version": "v1.25.10",
 "available_talos_version": "v1.4.4",
 "needs_kubernetes_update": true,
 "outdated_talos_versions": [],
 "needs_talos_update": false,
 "status": "NEEDS-KUBE-UPDATE-TO-v1.25.10-FROM-v1.27.1"
}
```

It's picking K8S 1.25.10, because that's the version marked as "latest" here https://github.com/kubernetes/kubernetes/releases , but I need to patch this. Should this point to the "biggest" version? Biggest within the major version?)

**Reply by smira** [reply, marked as answer]
There is a support matrix for each release of Talos: https://www.talos.dev/v1.4/introduction/support-matrix/

Also the same matrix is codified in the machinery package, if you're looking for automation.

In general, we recommend upgrading to Talos point releases, as they contain fixes. And jumping to the next minor release after some testing.

Kubernetes should stay within n-2 version support guarantee. Same way, deploy patch versions faster, minor versions with some testing.

Order of operations doesn't matter as long as you're within the supported release combination.

**Reply by alexpdp7** [reply]
Hi, I propose https://github.com/siderolabs/talos/pull/7285 to make this a bit more discoverable. But the rest is fairly intuitive, so I think not many more docs are needed.

I think I'll mirror that information in my monitor. It would be nice to have this info published to a static URL as JSON, or have a `talosctl` command that prints it out as JSON. But it's really nice-to-have, I don't think this information updates so frequently.

**Reply by alexpdp7** [reply]
OK, got https://github.com/alexpdp7/talos-check working on my two clusters, using Nagios. Now I get a warning:

```
WARNING Running outdated Kubernetes version v1.27.1, update to v1.27.2. Running Talos v1.4.4: available v1.4.4. Running Kubernetes v1.27.1, available v1.27.2. 
```

It picks K8S versions fairly aggressively, but what the "deployment" exposes is pretty generic:

```
{
 "talos_versions": [
  "v1.4.4"
 ],
 "available_talos_version": "v1.4.4",
 "kubernetes_version": "v1.27.1",
 "latest_available_supported_kubernetes_versions": {
  "1.27": "1.27.2",
  "1.26": "1.26.5",
  "1.25": "1.25.10"
 }
}
```

, so you could really implement other update policies.

... now, on to update Kubernetes...
